ci/mutmut_report.py

```diff
@@ -0,0 +1,48 @@
+import subprocess
+import json
+import os
+import csv
+
+
+def create_report():
+    mutmut_result = subprocess.check_output(["mutmut", "results"])
+    survival_rate = extract_survival_rate(mutmut_result)
+    dict_report = create_report_dict(survival_rate)
+
+    write_json_report(dict_report)
+    write_csv_report(dict_report)
+
+
+def extract_survival_rate(mutmut_result):
+    lines: list = mutmut_result.decode().split("\n")
+    before_survivors = [i for i, s in enumerate(lines) if "Survived" in s]
+    del lines[0:before_survivors[0]+1]
+
+    return sum("mutmut apply" in l for l in lines)
+
+
+def create_report_dict(survival_rate):
+    return {"survival_rate": survival_rate}
+
+
+def write_json_report(dict_report):
+    filename = '.mutmut/mutmut_report.json'
+    os.makedirs(os.path.dirname(filename), exist_ok=True)
+    with open(filename, "w") as f:
+        f.write(json.dumps(dict_report))
+        f.close()
```

Comment:
I think that 'with' closes the file for you?

Reply:
Indeed it does.